File: app/src/main/java/com/shullock/app/data/model/Shul.kt
package com.shullock.app.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "shuls")
data class Shul(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val radiusMeters: Float = 100f,
    val address: String = "",
    val isEnabled: Boolean = true
)

File: src/types/Shul.ts
export interface Shul {
  id: string;
  name: string;
  latitude: number;
  longitude: number;
  radiusMeters: number;
  address: string;
  isEnabled: boolean;
}

export interface ShulLockState {
  isActive: boolean;
  activeShulName: string | null;
  activeShulId: string | null;
}

export interface AppSettings {
  autoActivate: boolean;
  autoDeactivate: boolean;
  startupMode: boolean;
}

export interface SearchResult {
  address: string;
  latitude: number;
  longitude: number;
}
